pkg/commands/compute/deploy_test.go

```diff
@@ -225,6 +232,98 @@ func TestDeploy(t *testing.T) {
 				"Creating service...",
 			},
 		},
+		// The following test mocks the service creation to fail with a specific
```

Comment:
Here are a bunch of new tests that validate different failure and success scenarios related to the trial activation.